packages/shared/src/components/Sidebar.tsx

```diff
@@ -1,105 +1,173 @@
+import SearchIcon from '../../icons/magnifying.svg';
+import DisqusIcon from '../../icons/comment.svg';
+import BookmarkIcon from '../../icons/bookmark.svg';
+import EyeIcon from '../../icons/eye.svg';
+import SettingsIcon from '../../icons/settings.svg';
 import { SimpleTooltip } from './tooltips/SimpleTooltip';
 
-const asideWidth = sizeN(89);
+interface SidebarMenuItems {
+  key: string;
+  items: SidebarMenuItem[];
+}
+
+interface SidebarMenuItem {
+  icon: ReactElement;
+  title: string;
+  path?: string;
+  action?: () => unknown;
+}
+
+const ListIcon = ({ Icon }): ReactElement => <Icon className="mr-3 w-5 h-5" />;
 
 export default function Sidebar(): ReactElement {
-  const { alerts, updateAlerts } = useContext(AlertContext);
-  const { feedSettings } = useFeedSettings();
-  const [opened, setOpened] = useState(false);
-  const { onboardingStep, incrementOnboardingStep } =
-    useContext(OnboardingContext);
-  const hightlightTrigger = onboardingStep === 2;
+  const { openSidebar, toggleOpenSidebar } = useContext(SettingsContext);
+  const [showSettings, setShowSettings] = useState(false);
+
+  const topMenuItems: SidebarMenuItems[] = [
+    {
+      key: 'Discover',
+      items: [
+        {
+          icon: <ListIcon Icon={FilterIcon} />,
+          title: 'Feed filters',
+          path: '/sidebar',
+        },
+        {
+          icon: <ListIcon Icon={HotIcon} />,
+          title: 'Popular',
+          path: '/popular',
+        },
+        {
+          icon: <ListIcon Icon={UpvoteIcon} />,
+          title: 'Most upvoted',
+          path: '/upvoted',
+        },
+        {
+          icon: <ListIcon Icon={DisqusIcon} />,
+          title: 'Best discussions',
+          path: '/discussed',
+        },
+        {
+          icon: <ListIcon Icon={SearchIcon} />,
+          title: 'Search',
+          path: '/search',
+        },
+      ],
+    },
+    {
+      key: 'Manage',
+      items: [
+        {
+          icon: <ListIcon Icon={BookmarkIcon} />,
+          title: 'Bookmarks',
+          path: '/bookmarks',
+        },
+        {
+          icon: <ListIcon Icon={EyeIcon} />,
+          title: 'Reading history',
+          path: '/history',
+        },
+        {
+          icon: <ListIcon Icon={SettingsIcon} />,
+          title: 'Customize',
+          action: () => setShowSettings(!showSettings),
+        },
+      ],
+    },
+  ];
 
-  const toggleSidebar = () => {
-    if (opened) {
-      setOpened(false);
-    } else {
-      setOpened(true);
-      if (hightlightTrigger) {
-        incrementOnboardingStep();
-      }
-      if (
-        alerts?.filter &&
-        (feedSettings?.includeTags?.length ||
-          feedSettings?.blockedTags?.length ||
-          feedSettings?.excludeSources?.length ||
-          feedSettings?.advancedSettings?.length)
-      ) {
-        updateAlerts({ filter: false });
-      }
-    }
-  };
+  const ItemInner = ({ item }: { item: SidebarMenuItem }) => (
```

Comment:
This can go outside the component

Reply:
Then we should pass openSidebar to it, that's the only reason it's sitting inside now.
Not sure what's the better one here in terms of performance?

Reply:
My personal preference is to take out everything that has little to no use in the closure and scope of the parent.
Make everything more readable.